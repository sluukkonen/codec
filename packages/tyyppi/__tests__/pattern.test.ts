import { string, pattern } from "../src/index.js"
import { expectParseFailure, expectParseSuccess } from "./helpers.js"

const whitespace = pattern(string, /^\s*$/)

test("should parse strings matching the pattern", () => {
  expectParseSuccess(whitespace, "")
  expectParseSuccess(whitespace, "   ")
})

test("should reject strings not matching the pattern", () => {
  expectParseFailure(whitespace, "abc")
})

test("should reject non-strings", () => {
  expectParseFailure(whitespace, null)
})
