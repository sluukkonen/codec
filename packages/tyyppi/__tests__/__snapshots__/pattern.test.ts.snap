// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should reject non-strings 1`] = `
[
  {
    "code": "invalid_type",
    "expected": "string",
    "message": "Expected a string, received null",
    "path": [],
    "received": "null",
  },
]
`;

exports[`should reject strings not matching the pattern 1`] = `
[
  {
    "code": "invalid_pattern",
    "message": "Expected the string to match the pattern /^\\s*$/",
    "path": [],
    "pattern": "/^\\s*$/",
  },
]
`;
